pinecone/control/pinecone.py

```diff
@@ -544,6 +544,33 @@ def describe_index(self, name: str):
 
         return IndexModel(description)
 
+    def has_index(self, name: str) -> bool:
+        """Checks if a Pinecone index exists.
+
+        :param name: The name of the index to check for existence.
+        :return: Returns `True` if the index exists, `False` otherwise.
+
+        ### Example Usage
+
+        ```python
+        from pinecone import Pinecone
+
+        client = YourClass()  # Initialize your class instance
+        exists = client.has_index("my_index_name")
+
+        if exists:
+            print("The index exists.")
+        else:
+            print("The index does not exist.")
+        ```
+        """
+
+        try:
+            self.index_api.describe_index(name)
+            return True
+        except Exception as e:
+            return False
```

Comment:
This should work most of the time, but now that I'm thinking about it more I realized that exceptions can occur that should not be ignored. For example, a 500 error coming from the backend should throw rather than be ignored. Or if you try calling with an incorrect api key, that's a 401 unauthorized that the user should see an error about.

It would probably be simpler to call `list_indexes` and check if the index is in the list instead of ignoring exceptions.

Reply:
Agreed, this is definitely the downside of the former approach:
> For example, a 500 error coming from the backend should throw rather than be ignored. Or if you try calling with an incorrect api key, that's a 401 unauthorized that the user should see an error about.